xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -264,12 +266,24 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
         localXdsClientWrapperForServerSds =
             XdsClientWrapperForServerSds.newInstance(port, Bootstrapper.getInstance(), syncContext);
       } catch (Exception e) {
-        logger.log(Level.WARNING, "Exception while creating the xDS client", e);
+        logger.log(Level.FINE, "Fallback to plaintext due to exception", e);
         localXdsClientWrapperForServerSds = null;
       }
       this.xdsClientWrapperForServerSds = localXdsClientWrapperForServerSds;
     }
 
+    @VisibleForTesting
+    public ServerSdsProtocolNegotiator(
+        DownstreamTlsContext downstreamTlsContext,
+        XdsClientWrapperForServerSds xdsClientWrapperForServerSds) {
+      this.downstreamTlsContext = downstreamTlsContext;
+      this.xdsClientWrapperForServerSds = xdsClientWrapperForServerSds;
```

Comment:
nit: null check here seems reasonable

Reply:
with the refactoring both params can be null (plaintext case) so no null checks

Reply:
then add `@Nullable`, if both are null isn't it plaintext?